fedimint-core/src/encoding/mod.rs

```diff
@@ -966,15 +1063,17 @@ impl<T> Decodable for DynRawFallback<T>
 where
     T: Decodable + 'static,
 {
-    fn consensus_decode<R: std::io::Read>(
+    fn consensus_decode_from_finite_reader<R: std::io::Read>(
         reader: &mut R,
         decoders: &ModuleDecoderRegistry,
     ) -> Result<Self, crate::encoding::DecodeError> {
         let module_instance_id =
-            fedimint_core::core::ModuleInstanceId::consensus_decode(reader, decoders)?;
+            fedimint_core::core::ModuleInstanceId::consensus_decode_from_finite_reader(
+                reader, decoders,
+            )?;
         Ok(match decoders.get(module_instance_id) {
             Some(decoder) => {
-                let total_len_u64 = u64::consensus_decode(reader, decoders)?;
+                let total_len_u64 = u64::consensus_decode_from_finite_reader(reader, decoders)?;
                 let mut reader = reader.take(total_len_u64);
                 let v: T = decoder.decode(&mut reader, module_instance_id, decoders)?;
```

Comment:
This value being ridiculously high won't be a problem because the underlying reader is already limited.